cinder/client.go

```diff
@@ -0,0 +1,264 @@
+package cinder
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type TokenFn func() string
+
+func SetEndpointFn(endpoint *url.URL, wrappedHandler RequestHandlerFn) RequestHandlerFn {
```

Comment:
Documentation corrected.

Regarding the endpoint parameter, I had this same debate with myself. The reason I went with URL is because I wanted a well-formed host, and URL guarantees this. I could take in a string and validate in this function, but that breaks the desired signature by returning an error (this makes it less "nice" to chain wrapper functions).

It's definitely debatable... given that new information, do you have any opinions/suggestions?